reflex/utils/prerequisites.py

```diff
@@ -265,75 +266,73 @@ def download_and_run(url: str, *args, show_status: bool = False, **env):
     show(f"Installing {url}", process)
 
 
-def download_and_extract_fnm_zip(url: str):
+def download_and_extract_fnm_zip():
     """Download and run a script.
 
-    Args:
-        url: The url of the fnm release zip binary.
-
     Raises:
         Exit: If an error occurs while downloading or extracting the FNM zip.
     """
-    # TODO: make this OS agnostic
     # Download the zip file
+    url = constants.FNM_INSTALL_URL
     console.debug(f"Downloading {url}")
-    fnm_zip_file = f"{constants.FNM_DIR}\\fnm_windows.zip"
-    # Function to download and extract the FNM zip release
+    fnm_zip_file = os.sep.join([constants.FNM_DIR, f"{constants.FNM_FILENAME}.zip"])
+    # Function to download and extract the FNM zip release.
     try:
-        # Download the FNM zip release
+        # Download the FNM zip release.
         # TODO: show progress to improve UX
         with httpx.stream("GET", url, follow_redirects=True) as response:
             response.raise_for_status()
             with open(fnm_zip_file, "wb") as output_file:
                 for chunk in response.iter_bytes():
                     output_file.write(chunk)
 
-        # Extract the downloaded zip file
+        # Extract the downloaded zip file.
         with zipfile.ZipFile(fnm_zip_file, "r") as zip_ref:
             zip_ref.extractall(constants.FNM_DIR)
 
-        console.debug("FNM for Windows downloaded and extracted successfully.")
+        console.debug("FNM package downloaded and extracted successfully.")
     except Exception as e:
         console.error(f"An error occurred while downloading fnm package: {e}")
         raise typer.Exit(1) from e
     finally:
-        # Clean up the downloaded zip file
+        # Clean up the downloaded zip file.
         path_ops.rm(fnm_zip_file)
 
 
 def install_node():
-    """Install nvm and nodejs for use by Reflex.
+    """Install fnm and nodejs for use by Reflex.
     Independent of any existing system installations.
     """
-    if constants.IS_WINDOWS:
-        path_ops.mkdir(constants.FNM_DIR)
-        if not os.path.exists(constants.FNM_EXE):
-            download_and_extract_fnm_zip(constants.FNM_WINDOWS_INSTALL_URL)
+    if not constants.FNM_FILENAME:
+        # fnm only support Linux, macOS and Windows distros.
+        return
```

Comment:
so what will happen in this case? can the user install and run their own copy of `node` in these environments? asking because it looks like we set `NODE_BIN_PATH` in an FNM-specific way

Reply:
Thats a good question, Previously(when users had to install node themselves), we checked for node on the host machine (`which node` or use the node command and check for errors) and used that. We can do that as well as expose a config knob (just like we do for bun) to allow users specify custom paths to node binaries, but Im open to suggestions. What do you think? 
cc: @masenf @picklelo @Lendemor @jackie-pc

Reply:
the `which` solution sounds reasonable to me.  Should we log something (even if debug level) when we fall through to such cases.  I.e. either we use `which node` or straight give up.  Note this should be a pretty rare case and so it's worth logging something.  I.e. we are talking about an environment that is NOT Windows, Linux or Mac (roughly)